### DISCUSSION TITLE
Use of shouldUseNativeValidation and messaging

### DISCUSSION THREAD

**Discussion by qasimalyas** [react-hook-form/react-hook-form#6128] [General] [answered]
I was testing the `shouldUseNativeValidation` feature and wanted to ask question about it. 

From looking at the code examples on this page https://react-hook-form.com/api/useform it looks as though you can just pass the mentioned option and it should work. I've created and example https://codesandbox.io/s/romantic-shirley-mmt3q?file=/src/App.js and set `required: true` but am getting a blank error message on submission. I was expecting the browser to return the default messaging in the client locale. If I pass in string `required: 'my validation message` then this shows fine. 

My question is that is there any way to let browser supply the validation message (which is useful to handle various locales).

**Comment by bluebill1049** [comment, marked as answer]
do this instead if you don't need a custom message

```tsx
import React from "react";
import { useForm } from "react-hook-form";

export default function App() {
  const { register, handleSubmit } = useForm({
    shouldUseNativeValidation: true
  });
  const onSubmit = async (data) => {
    console.log(data);
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input
        {...register("firstName")}
        required
      />
      <br />
      <br />
      <input type="submit" />
    </form>
  );
}

```

**Reply by qasimalyas** [reply]
Thanks that did it!
